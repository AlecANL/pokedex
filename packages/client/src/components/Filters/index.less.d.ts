// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'filters-component': string;
}
declare var cssExports: CssExports;
export = cssExports;
