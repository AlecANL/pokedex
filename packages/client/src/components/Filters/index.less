.filters-component {
  width: 80%;
}
