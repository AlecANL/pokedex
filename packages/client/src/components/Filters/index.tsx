import { Col, Input, Row, Select } from 'antd';
import React from 'react';
import { formatMessage } from 'umi-plugin-react/locale';
import style from './index.less';

interface IFiltersProps {
  classifications: Array<PokemonClassification>;
  types: Array<PokemonType>;
  onSearch: (filter: { [key: string]: string }) => void;
}

const { Search } = Input;

const { Option } = Select;

/**
 * This component show all available options in order to filter all available options
 *
 * @param props
 */

type FilterType = 'classification' | 'type' | 'name';

const Filters: React.FC<IFiltersProps> = props => {
  const onSearch = (value: string, key: FilterType) => {
    // const criteria = { [key]: value };
    // props.onSearch(_.omitBy(criteria, _.isNil));
    props.onSearch({ [key]: value });
  };

  return (
    <div className={style['filters-component']}>
      <Row type="flex" gutter={[16, 16]}>
        <Col span={8}>
          <Select
            onChange={(value: string) => onSearch(value, 'classification')}
            showSearch
            defaultValue=""
            style={{ width: '100%' }}
          >
            <Option value="">
              {formatMessage({ id: 'components.filters.allClassification' })}
            </Option>
            {props.classifications.map(classification => {
              return <Option value={classification}>{classification}</Option>;
            })}
          </Select>
        </Col>

        <Col span={8}>
          <Select
            onChange={(value: string) => onSearch(value, 'type')}
            showSearch
            defaultValue=""
            style={{ width: '100%' }}
          >
            <Option value="">{formatMessage({ id: 'components.filters.allTypes' })}</Option>
            {props.types.map(type => {
              return <Option value={type}>{type}</Option>;
            })}
          </Select>
        </Col>

        <Col span={8}>
          <Search
            onSearch={(value: string) => onSearch(value, 'name')}
            enterButton
            style={{ width: '100%' }}
          />
        </Col>
      </Row>
    </div>
  );
};

export default Filters;
