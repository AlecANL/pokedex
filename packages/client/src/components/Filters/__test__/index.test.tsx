import React from 'react';
import renderer from 'react-test-renderer';
import Filters from '..';

jest.mock('umi-plugin-react/locale');

describe('Components/Filters', () => {
  it('Get snapshot of filters', () => {
    const component = renderer.create(
      <Filters classifications={[]} types={[]} onSearch={() => true} />,
    );
    let snap = component.toJSON();
    expect(snap).toMatchSnapshot();
  });
});
