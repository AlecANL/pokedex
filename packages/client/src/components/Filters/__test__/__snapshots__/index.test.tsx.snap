// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Filters Get snapshot of filters 1`] = `
<div
  className="filters-component"
>
  <div
    className="ant-row-flex"
    style={
      Object {
        "marginBottom": -8,
        "marginLeft": -8,
        "marginRight": -8,
        "marginTop": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title=""
            />
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title=""
            />
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <span
        className="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <input
            className="ant-input"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-group-addon"
          >
            <button
              className="ant-btn ant-input-search-button ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                aria-label="icon: search"
                className="anticon anticon-search"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </i>
            </button>
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
